internal/authz/authz.go

```diff
@@ -222,6 +308,45 @@ func (a *Authorizer) individualAuthorize(ctx context.Context, clientID, relation
 	return nil
 }
 
+// moduleAuthorize checks if the user has access to each of the modules, and exits if an error is encountered.
+func (a *Authorizer) moduleAuthorize(ctx context.Context, clientID, relation, storeID string, modules []string) error {
```

Comment:
Since these involves go func - ideally, we have unit test on this function (with verification on no go-leak)
```
	t.Cleanup(func() {
		goleak.VerifyNone(t)
	})
```